effekt/shared/src/main/scala/effekt/machine/Transformer.scala

```diff
@@ -201,7 +200,21 @@ object Transformer {
   def transform(arg: lifted.Argument)(using BlocksParamsContext, Context): Binding[Variable] = arg match {
     case expr: lifted.Expr => transform(expr)
     case block: lifted.Block => transform(block)
-    case lifted.Evidence(_) => transform(lifted.IntLit(0)) // TODO implement
+    case lifted.Evidence(scopes) => {
+      val addSymbol = Context.module.findPrelude.terms("infixAdd").find({
```

Comment:
In my opinion, we should have a separate type of evidence in `machine.Type` and two runtime operations in `machine.Statement`: `LiteralEvidence(n: Int)` and `ComposeEvidence(name: Variable, arguments: Environment, rest: Statement)`.

Reply:
At first we didn't want to add evidence to reduce the number of cases. What are the advantages you hope from having evidence being built into machine?